New question logic

leenagupte opened the pull request:
Trello: 
- https://trello.com/c/0flnhchV
- https://trello.com/c/81u4ftj7

Review app 👉 https://coronavirus-new-questio-ustwsp.herokuapp.com/live-in-england

## What's changed?

Stephen G has created a new question logic that will reduce the number of steps for users and show different confirmation messages for different outcomes, meaning content can be more relevant to a user's circumstances.

Spike this approach (the `27 April` tab on the [questions protocol](https://docs.google.com/spreadsheets/d/1o0C0NabtFykGLpCXuDD_rRkKG9tAI4AVzQCMPeKx3ko/edit#gid=19251621)) to get an understanding of complexity, impact on the downstream data store and our analytics.

## Why?

 Adding routing to allow users to skip the medical conditions question if we don’t them to answer it simplifies the form for the majority of users. 

The idea behind having two different versions of the confirmation page is that the user only sees the version that says “contact your GP" if they haven't already had contact from their GP. The NHS wants to stop GPs being contacted unnecessarily.

## Expected changes
- Content changes to the NHS Letter page
- Medical conditions question is skipped if the user answers "Yes..." to NHS letter page
- Two variants of confirmation page:
  -- Contact GP (user selected "No...." or "Not sure" to NHS letter question)
  -- No further action (user selected "Yes..." to NHS letter question)
- Two variants of confirmation email:
  -- Contact GP (user selected "No...." or "Not sure" to NHS letter question)
  -- No further action (user selected "Yes..." to NHS letter question)
- Two variants of confirmation sms:
  -- Contact GP (user selected "No...." or "Not sure" to NHS letter question)
  -- No further action (user selected "Yes..." to NHS letter question)


### NHS letter page
<img width="1552" alt="Screenshot 2020-05-05 at 16 53 12" src="https://user-images.githubusercontent.com/5793815/81088254-a2a25980-8ef2-11ea-85e2-a333ca06f95b.png">

### Check answers page
#### Without medical conditions question
<img width="1552" alt="Screenshot 2020-05-05 at 16 52 39" src="https://user-images.githubusercontent.com/5793815/81088243-9fa76900-8ef2-11ea-992f-df093b32aa10.png">

#### With medical conditions question
<img width="1552" alt="Screenshot 2020-05-05 at 16 53 05" src="https://user-images.githubusercontent.com/5793815/81088256-a33af000-8ef2-11ea-8f57-f7fe574e073e.png">

### Confirmation page
#### Contact GP
![confirmation_page_contact_gp](https://user-images.githubusercontent.com/5793815/82565885-0c696700-9b73-11ea-9240-da3185bf9fa8.png)

#### No further action required
![confirmation_page_no_action](https://user-images.githubusercontent.com/5793815/82565781-e0e67c80-9b72-11ea-8d9e-6f16fc8d6eb1.png)

### Confirmation email
#### Contact GP
![confirmation_email_contact_gp](https://user-images.githubusercontent.com/5793815/82565954-26a34500-9b73-11ea-89c7-5ba62ca63f17.png)

#### No further action required
![confirmation_email_no_action](https://user-images.githubusercontent.com/5793815/82565981-2efb8000-9b73-11ea-9242-d7a3797a8a1b.png)

### Confirmation SMS
#### Contact GP
![confirmation_sms_contact_gp](https://user-images.githubusercontent.com/5793815/82566047-433f7d00-9b73-11ea-86f2-1a5b6eb1c47a.jpg)

#### No further action required
![confirmation_no_action_sms](https://user-images.githubusercontent.com/5793815/82566074-4e92a880-9b73-11ea-9c26-aa623f19eba2.jpg)

## How to test locally
Follow the instructions here to be able to test email and sms locally: https://github.com/alphagov/govuk-coronavirus-vulnerable-people-form/pull/309/commits/f0d3fbd8afad38ecc801c9e80b625deb0db796c8

leenagupte committed:
Update NHS letter content

leenagupte committed:
Skip medical conditions question if has letter

leenagupte committed:
Update medical conditions page content

leenagupte committed:
Add condition to previous path on name page

If the user was allowed to skip the medical conditions question, they should be directed to the nhs letter question.

leenagupte committed:
Update content on confirmation page

leenagupte committed:
Remove hack for the medical conditions question

Trello: https://trello.com/c/81u4ftj7

We're back to having only a "Yes" or "No" option, so this hack is no longer needed.

leenagupte committed:
Restructure confirmation description

This is a preparatory step for the next commit where alternative text for the confirmation will be added.

leenagupte committed:
Add alternative confirmation content

leenagupte committed:
Pass a further_actions querystring param to confirmation page

This will allow the confirmation page to determine which description to display.

leenagupte committed:
Display alternate confirmation message

Display an alternate confirmation message telling the user to contact their GP if the user didn't receive an NHS letter.

leenagupte committed:
Don't show medical_conditions on check answer page

Don't show the medical conditions question on the check answer page if the user skipped the question.

leenagupte committed:
Clear medical conditions value if nhs letter is yes

Clear the medical_conditions session value if the user changes their response to the nhs_letter questions from No to Yes.
This will stop the medical conditions question appearing on the check answers page.

leenagupte committed:
Force users to answer the medical conditions question

Force users to answer the medical conditions if they are not allowed to skip the question.
This will catch users who changed their answer to the nhs letter question from yes to no, and so had previously
skipped the medical conditions question.

bilbof commented on app/controllers/coronavirus_form/check_answers_controller.rb:
```diff
@@ -31,7 +26,7 @@ def submit
     reset_session
 
-    redirect_to confirmation_url(reference_number: submission_reference)
+    redirect_to confirmation_url(reference_number: submission_reference, contact_gp: contact_gp)
```
Could this be `contact_gp?` and you could remove the variable assignment above.

leenagupte replied:
No, because the session is reset right above this line, so the session data is already gone.

bilbof replied:
Ah I didn't notice that 👍

bilbof reviewed (approved):
Looks great!

leenagupte committed:
Restructure email body text

This is a preparatory step for the next commit where alternative text for the email will be added.

leenagupte committed:
Add alternative email content for contacting GP

leenagupte committed:
Send email with alternative body text

leenagupte committed:
Update "no further action" confirmation page content

leenagupte committed:
Restructure sms body text

This adds new content as well as preparing for the next commit where alternative text for the sms will be added.

leenagupte committed:
Add alternative sms content

leenagupte committed:
Send sms with alternative body text

leenagupte committed:
Update testing email and sms locally docs

bilbof reviewed (approved):
Great one  👏 

I also learnt about a new Star Wars character from this 😄
